### PR TITLE
Approximate axis dimensions for SSR; add SSR checks for unsupported calls

### PR DESCRIPTION
**Pull request by marjan-georgiev**
**What kind of change does this PR introduce?** (check one with "x")
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

Using the charts in SSR throws an error when some of the non-supported methods are called.

**What is the new behavior?**

Charts work in SSR

**Does this PR introduce a breaking change?** (check one with "x")
- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: projects/swimlane/ngx-charts/src/lib/common/axes/x-axis-ticks.component.ts [hunk 1]
```diff
@@ -79,10 +85,16 @@ export class XAxisTicksComponent implements OnChanges, AfterViewInit {
...
+      // for SSR, use approximate value instead of measured
+      this.dimensionsChanged.emit({ height: this.approxHeight });
+      return;
+    }
+
     const height = parseInt(this.ticksElement.nativeElement.getBoundingClientRect().height, 10);
     if (height !== this.height) {
       this.height = height;
-      this.dimensionsChanged.emit({ height });
+      this.dimensionsChanged.emit({ height: this.height });
```
Comment (Hypercubed) [inline projects/swimlane/ngx-charts/src/lib/common/axes/x-axis-ticks.component.ts:97]: Is this needed?

**Review by Hypercubed — APPROVED** [review]
Code LGTM... haven't tested.

File: projects/swimlane/ngx-charts/src/lib/common/axes/x-axis-ticks.component.ts (hunk 1 above)
Comment (marjan-georgiev) [reply projects/swimlane/ngx-charts/src/lib/common/axes/x-axis-ticks.component.ts:97]: Thought it's cleaner to emit the actual property value.

**Review by nelsonGuti — APPROVED** [review]
LGTM

**Pull request merged**